src/core/initializeComponentsFactory.js

```diff
@@ -30,16 +34,15 @@ export default (
   config.validate();
   componentCreators.forEach(createComponent => {
     const { namespace } = createComponent;
-    const { propertyID, cookieDomain } = config;
-    const storage = getNamespacedStorage(config.orgID);
     // TO-DOCUMENT: Helpers that we inject into factories.
     let component;
     try {
       component = createComponent({
         logger: logger.spawn(`[${namespace}]`),
         cookie: cookie(namespace, propertyID, cookieDomain),
         config,
-        storage
+        storage,
+        enableOptIn: optIn.enable
```

Comment:
This is kind of a key point. This is how Privacy is able to configure the `optIn` object without any component getting access to the rest of the `optIn` object first.